### PR TITLE
Add error message for illegal literals - Parser.scala:602

### PR DESCRIPTION
**Pull request by maseev**
This pull request is related to the following [issue](https://github.com/lampepfl/dotty/issues/1589)

### CODE REVIEW

File: src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 1]
```diff
@@ -754,4 +754,20 @@ object messages {
...
+      hl"""|Expected a valid literal.
+           |
+           |Available literals can be divided into the several groups:
+           | - Integer literals: 0, 21, 0xFFFFFFFF, -42L
+           | - Floating Point Literals: 0.0, 1e30f, 3.14159f, 1.0e-100, .1
+           | - Boolean Literals: true, false
+           | - Character Literals: 'a', '\u0041', '\\n'
+           | - String Literals: "Hello, World!"
+           | - null
+           |""".stripMargin
```
Comment (felixmulder) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: Strip margin isn't necessary on `hl` strings :)

File: src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 2]
```diff
@@ -754,4 +754,20 @@ object messages {
...
+    val kind = "Syntax"
+    val msg = "illegal literal"
+    val explanation =
+      hl"""|Expected a valid literal.
+           |
+           |Available literals can be divided into the several groups:
+           | - Integer literals: 0, 21, 0xFFFFFFFF, -42L
+           | - Floating Point Literals: 0.0, 1e30f, 3.14159f, 1.0e-100, .1
+           | - Boolean Literals: true, false
+           | - Character Literals: 'a', '\u0041', '\\n'
```
Comment (felixmulder) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: There shouldn't be an extra `\` in `'\\n'` right? You don't need to escape things (except `$`) in triple quoted strings.

File: src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 3]
```diff
@@ -754,4 +754,20 @@ object messages {
            |${"square(ints: _*)          // res1: List[Int] = List(4, 9, 16)"}
            |""".stripMargin
   }
+
+  case class IllegalLiteral()(implicit ctx: Context) extends Message(28) {
+    val kind = "Syntax"
+    val msg = "illegal literal"
+    val explanation =
+      hl"""|Expected a valid literal.
```
Comment (felixmulder) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: I would skip this line, the error already says "illegal literal" right? If you think it should be kept, you could bake it into the sentence where you present the legal literals in Scala.

**Comment by maseev** [comment]
@felixmulder PTAL

**Comment by felixmulder** [comment]
LGTM

File: src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 4]
```diff
@@ -754,4 +754,18 @@ object messages {
            |${"square(ints: _*)          // res1: List[Int] = List(4, 9, 16)"}
            |""".stripMargin
   }
+
+  case class IllegalLiteral()(implicit ctx: Context) extends Message(28) {
+    val kind = "Syntax"
+    val msg = "illegal literal"
+    val explanation =
+      hl"""|Available literals can be divided into the several groups:
```
Comment (smarter) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: typo: "into the several groups" -> "into several groups"

Comment (felixmulder) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: Fixed in: bffefe96f1c3c550e42d745af879755d1f99bc37

Comment (maseev) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: Oh, snap! Sorry for this typo guys.

**Pull request merged**